primitives/xcm-commons/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use staging_xcm::latest::prelude::*;
+trait Parse {
+    /// Returns the "chain" location part. It could be parent, sibling
+    /// parachain, or child parachain.
+    fn chain_part(&self) -> Option<Location>;
+}
+
+impl Parse for Location {
+    fn chain_part(&self) -> Option<Location> {
+        match (self.parents, self.first_interior()) {
+            // sibling parachain
+            (1, Some(Parachain(id))) => Some(Location::new(1, [Parachain(*id)])),
+            // parent
+            (1, _) => Some(Location::parent()),
```

Comment:
We should re-check these variants for the case of dancelight. Given that it's a relay context, not sure if we want to return something for "parent" or "sibling parachains".

Should we create a new trait/struct for dancelight? What do you think @girazoki @dimartiro?

Reply:
I think the goal is to re-use them for both sibling parachain and relay mode. So I think it's ok to keep them